### DISCUSSION TITLE
About the atom type embedding algorithm

### DISCUSSION THREAD

**Discussion by dxy-cs** [deepmodeling/deepmd-kit#1711] [Q&A]
The file deepmd/doc/development/type_embedding.md indicates that the atom type embedding algorithm will be executed if there is “type_embedding” section in the input.json, which can make the embedding net and fitting net shard. And the input parameter "type_one_side" decides whether the information of the central atom will be used in building the descriptor.
Here, I have a question:  
I find that there's not "type_embedding" section in the input.json of the three dataset provided by ASC22. However, the “type_one_side” is set as changable. And from the results we've got, the "type_one_side" does make an difference to the training process.  That's mean, even though we didn't have the "type_embedding" section, the "type_one_side" still have an effect. Then, whether the atom type embedding algorithm got executed in the asc22-branch? If so, what's the meaning of existing about "type_embedding" section?

**Comment by njzjz** [comment]
No, “type_embedding” decides whether the information of the central atom will be used in building the descriptor. "type_one_side" makes the embedding net shared.

**Reply by dxy-cs** [reply]
**If "type_one_side" is TRUE, try to build N_types embedding nets. Otherwise, building N_types^2 embedding nets.**
This is one of the statements about "type_one_side" in the official documents. I can understand the share of embedding nets decided by "type_one_side". 
The other statement about "type_one_side" is in deepmd/doc/development/type_embedding.md, which tells us "type_one_side" decides whether the information of the central atom will be used, in the case that "type_embedding" was detected.

So, whether the "type_one_side" plays different roles in case that  "type_embedding" is detected and is not detected? 

And,  what does the atom type embedding algorithm do on earth?  Does it make all different atom types share one(only one) embedding net and one fitting net?

**Reply by njzjz** [reply]
In the filter network, usually, N^2 networks are built. N networks will be built when `type_one_side` is true. When `type_embedding` is provided, it will be added as an extra input to the filter network. Here, `type_embeding` can also be used when `type_one_side` is false.

In the fitting network, usually, N networks are built. 1 network will be built when `type_embedding` is true. `type_embedding` will provide extra input to the fitting network.
